collectors/build/jenkins-audit/src/main/java/com/capitalone/dashboard/util/AuditDataCSVFileCreator.java

```diff
@@ -0,0 +1,275 @@
+            e.printStackTrace();
+        }
+    }
+
+    private void getAuditResults(Iterable<Dashboard> dashboards, long timestamp) {
+        List<AuditResult> auditResults = new ArrayList();
+        // CSV Header
+        entireCSVData.add(String.join(",", Arrays.asList(CSV_HEADER)));
+
+        dashboards.forEach(dashboard -> {
+            try {
+                DashboardReviewResponse dashboardReviewResponse = dashboardAuditService.getDashboardReviewResponse(
+                        dashboard.getTitle(), dashboard.getType(), dashboard.getConfigurationItemBusServName(),
+                        dashboard.getConfigurationItemBusAppName(), BEGIN_DATE, END_DATE, AUDIT_TYPE
+                );
+                AuditResult auditResult = new AuditResult(dashboard.getId(), dashboardReviewResponse, timestamp);
+
+                auditResults.add(auditResult);
+                this.assignAuditDataValues(auditResult);
+                this.addCSVRowData(dashboard);
+
+            } catch (Exception e) {
+                e.printStackTrace();
+                LOGGER.error("AUDIT API ERROR - " + e.getMessage());
+                Cmdb cmdb = cmdbRepository.findByConfigurationItem(dashboard.getConfigurationItemBusServName());
+                List<String> csvAuditAPIFailedRow = Arrays.asList(cmdb.getOwnerDept(), dashboard.getConfigurationItemBusServName(),
+                        dashboard.getConfigurationItemBusServName(), cmdb.getBusinessOwner(), cmdb.getAppServiceOwner(),
+                        "", "NO_DATA", "NA", "", "AUDIT API ERROR - " + e.getMessage());
+                entireCSVData.add(String.join(",", csvAuditAPIFailedRow));
+            }
+        });
+    }
+
+    private void addCSVRowData(Dashboard dashboard) {
+        Cmdb cmdb = cmdbRepository.findByConfigurationItem(dashboard.getConfigurationItemBusServName());
+        lob = cmdb.getOwnerDept();
+        asv = dashboard.getConfigurationItemBusServName();
+        bap = dashboard.getConfigurationItemBusServName();
+        asvOwner = cmdb.getBusinessOwner();
+        serviceOwner = cmdb.getAppServiceOwner();
+        url = "";
+        List<String> csvRowData = Arrays.asList(lob, asv, bap, asvOwner, serviceOwner, type, collectionStatus, auditStatus, error, url);
+        entireCSVData.add(String.join(",", csvRowData));
+    }
+
+    private void assignAuditDataValues(AuditResult auditResult) {
+
+        List<String> auditTypes = Arrays.asList(AUDIT_TYPES);
+        Set<DashboardAuditStatus> dashboardAuditStatuses = auditResult.getDashboardReviewResponse().getAuditStatuses();
+        Map<AuditType, Collection<AuditReviewResponse>> auditDetailReview = auditResult.getDashboardReviewResponse().getReview();
+
+        for (String auditType : auditTypes) {
+            type = auditType;
+
+            switch (type) {
+                case "LIBRARY_POLICY": {
+                    if (dashboardAuditStatuses.contains(DashboardAuditStatus.DASHBOARD_LIBRARY_POLICY_ANALYSIS_CONFIGURED)) {
```

Comment:
use StringUtils for conditions. Using StringUtils will take care of NPE's

Reply:
No Strings involved here. Checking the object type in set collections. Added null and empty collection checks though.